src/Component/CircleCollider.jl

```diff
@@ -45,121 +46,122 @@ module CircleColliderModule
+                        if this.currentRests[j].getParent().transform.getPosition().y == colliders[i].getParent().transform.getPosition().y && this.currentRests[j].getSize().y == colliders[i].getSize().y
                             push!(this.currentRests, colliders[i])
-                            for eventToCall in this.collisionEvents
-                                eventToCall()
-                            end
-                            #Begin to overlap, correct position
-                            transform.setPosition(Math.Vector2f(transform.getPosition().x, transform.getPosition().y - collision[2]))
-                            this.isGrounded = true
+                            break
                         end
-                        # end
-                        # if collision[1] == Below::ColliderLocation
-                        #     push!(this.currentCollisions, colliders[i])
-                        #     for eventToCall in this.collisionEvents
-                        #         eventToCall()
-                        #     end
-                        # end
                     end
                 end
-                for i in 1:length(this.currentRests)
-                    if CheckIfResting(this, this.currentRests[i])[1] == false
-                        deleteat!(this.currentRests, i)
-                        break
+                transform = this.getParent().transform
+                # if collision[1] == Top::CollisionDirection
+                #     push!(this.currentCollisions, colliders[i])
+                #     for eventToCall in this.collisionEvents
+                #         eventToCall()
+                #     end
+                #     #Begin to overlap, correct position
+                #     transform.setPosition(Math.Vector2f(transform.getPosition().x, transform.getPosition().y + collision[2]))
+                # end
+                # if collision[1] == Left::CollisionDirection
+                #     push!(this.currentCollisions, colliders[i])
+                #     for eventToCall in this.collisionEvents
+                #         eventToCall()
+                #     end
+                #     #Begin to overlap, correct position
+                #     transform.setPosition(Math.Vector2f(transform.getPosition().x + collision[2], transform.getPosition().y))
+                # end
+                # if collision[1] == Right::CollisionDirection
+                #     push!(this.currentCollisions, colliders[i])
+                #     for eventToCall in this.collisionEvents
+                #         eventToCall()
+                #     end
+                #     #Begin to overlap, correct position
+                #     transform.setPosition(Math.Vector2f(transform.getPosition().x - collision[2], transform.getPosition().y))
+                # end
+                # if collision[1] == Bottom::CollisionDirection
+                #this.isGrounded = collision[1]
```

Comment:
can this deadwood be removed?

Reply:
Yep! This is something I need to revisit, but it probably won't be for a while